<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    
    <!-- Always force latest IE rendering engine or request Chrome Frame -->
    <meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
    
    <!-- Use title if it's in the page YAML frontmatter -->
    <title>ReferenceSequence</title>
    
    <link href="/stylesheets/bootstrap.css" rel="stylesheet" type="text/css" /><link href="/stylesheets/bootstrap-theme.css" rel="stylesheet" type="text/css" />
    <script src="/javascripts/bootstrap.js" type="text/javascript"></script>
  </head>
  
  <body class="allele allele_reference_sequence">
    <div class="container">
      <div class="col-md-4">
	<h3><a href="/">ClinGen data model</a></h3>
	<h4><a href="/allele/">allele model</a></h4>
	<ul>
	  <li><a href="/allele/allele_name.html">AlleleName</a></li>
	  <li><a href="/allele/canonical_allele.html">CanonicalAllele</a></li>
	  <li><a href="/allele/canonical_allele_identifier.html">CanonicalAlleleIdentifier</a></li>
	  <li><a href="/allele/complex_canonical_allele.html">ComplexCanonicalAllele</a></li>
	  <li><a href="/allele/complex_nucleotide_canonical_allele.html">ComplexNucleotideCanonicalAllele</a></li>
	  <li><a href="/allele/complex_protein_canonical_allele.html">ComplexProteinCanonicalAllele</a></li>
	  <li><a href="/allele/gene.html">Gene</a></li>
	  <li><a href="/allele/gene_identifier.html">GeneIdentifier</a></li>
	  <li><a href="/allele/genomic_reference_sequence.html">GenomicReferenceSequence</a></li>
	  <li><a href="/allele/genomic_simple_allele.html">GenomicSimpleAllele</a></li>
	  <li><a href="/allele/locus_identifer.html">LocusIdentifer</a></li>
	  <li><a href="/allele/nucleotide_simple_allele.html">NucleotideSimpleAllele</a></li>
	  <li><a href="/allele/protein_reference_sequence.html">ProteinReferenceSequence</a></li>
	  <li><a href="/allele/protein_simple_allele.html">ProteinSimpleAllele</a></li>
	  <li><a href="/allele/reference_genome.html">ReferenceGenome</a></li>
	  <li><a href="/allele/reference_sequence.html">ReferenceSequence</a></li>
	  <li><a href="/allele/simple_allele.html">SimpleAllele</a></li>
	  <li><a href="/allele/simple_canonical_allele.html">SimpleCanonicalAllele</a></li>
	  <li><a href="/allele/simple_nucleotide_canonical_allele.html">SimpleNucleotideCanonicalAllele</a></li>
	  <li><a href="/allele/simple_protein_canonical_allele.html">SimpleProteinCanonicalAllele</a></li>
	  <li><a href="/allele/transcript_reference_sequence.html">TranscriptReferenceSequence</a></li>
	  <li><a href="/allele/transcript_simple_allele.html">TranscriptSimpleAllele</a></li>
	  <li><a href="/allele/">allele model</a></li>
	</ul>
      </div>
      <div class="col-md-8">
	<h1>ReferenceSequence</h1>
	<h2 id="definition">Definition</h2>

<p>A versioned sequence of nucleotide bases or amino acids.  </p>

<h2 id="scope-and-usage">Scope and Usage</h2>

<p>Reference sequences each have a versioned accession number: A short string key that can be used to look up the exact sequence, must be in a public database.  Any change to the sequence necessitates a change to the version.</p>

<p>ReferenceSequence is abstract: actual instances must be either a GenomicReferenceSequence, a TranscriptReferenceSequence, or a ProteinReferenceSequence.</p>

<p>The main usage of ReferenceSequence is as a base against which variation can be described.</p>

<h2 id="exclusions-and-limitations">Exclusions and Limitations</h2>

<p>Only a subset of the relationships between different subclasses of ReferenceSequence are modeled in version 1.0.   In particular, the relationship between GenomicReferenceSequence and TranscriptReferenceSequence is left unmodeled.  In actuality, the relationship between two sequences of these types would be created by aligning the TranscriptReferenceSequence to the GenomicReferenceSequence  (using a tool such as BLAT.) , and details of this mapping would be included in a class joining these two types of ReferenceSequence.</p>

      </div>
    </div>
  </body>
</html>
