<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    
    <!-- Always force latest IE rendering engine or request Chrome Frame -->
    <meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
    
    <!-- Use title if it's in the page YAML frontmatter -->
    <title>SimpleCanonicalAllele</title>
    
    <link href="/stylesheets/bootstrap.css" rel="stylesheet" type="text/css" /><link href="/stylesheets/bootstrap-theme.css" rel="stylesheet" type="text/css" />
    <script src="/javascripts/bootstrap.js" type="text/javascript"></script>
  </head>
  
  <body class="allele allele_simple_canonical_allele">
    <div class="container">
      <div class="col-md-4">
	<h3><a href="/">ClinGen data model</a></h3>
	<h4><a href="/allele/">allele model</a></h4>
	<ul>
	  <li><a href="/allele/allele_name.html">AlleleName</a></li>
	  <li><a href="/allele/canonical_allele.html">CanonicalAllele</a></li>
	  <li><a href="/allele/canonical_allele_identifier.html">CanonicalAlleleIdentifier</a></li>
	  <li><a href="/allele/complex_canonical_allele.html">ComplexCanonicalAllele</a></li>
	  <li><a href="/allele/complex_nucleotide_canonical_allele.html">ComplexNucleotideCanonicalAllele</a></li>
	  <li><a href="/allele/complex_protein_canonical_allele.html">ComplexProteinCanonicalAllele</a></li>
	  <li><a href="/allele/gene.html">Gene</a></li>
	  <li><a href="/allele/gene_identifier.html">GeneIdentifier</a></li>
	  <li><a href="/allele/genomic_reference_sequence.html">GenomicReferenceSequence</a></li>
	  <li><a href="/allele/genomic_simple_allele.html">GenomicSimpleAllele</a></li>
	  <li><a href="/allele/locus_identifer.html">LocusIdentifer</a></li>
	  <li><a href="/allele/nucleotide_simple_allele.html">NucleotideSimpleAllele</a></li>
	  <li><a href="/allele/protein_reference_sequence.html">ProteinReferenceSequence</a></li>
	  <li><a href="/allele/protein_simple_allele.html">ProteinSimpleAllele</a></li>
	  <li><a href="/allele/reference_genome.html">ReferenceGenome</a></li>
	  <li><a href="/allele/reference_sequence.html">ReferenceSequence</a></li>
	  <li><a href="/allele/simple_allele.html">SimpleAllele</a></li>
	  <li><a href="/allele/simple_canonical_allele.html">SimpleCanonicalAllele</a></li>
	  <li><a href="/allele/simple_nucleotide_canonical_allele.html">SimpleNucleotideCanonicalAllele</a></li>
	  <li><a href="/allele/simple_protein_canonical_allele.html">SimpleProteinCanonicalAllele</a></li>
	  <li><a href="/allele/transcript_reference_sequence.html">TranscriptReferenceSequence</a></li>
	  <li><a href="/allele/transcript_simple_allele.html">TranscriptSimpleAllele</a></li>
	  <li><a href="/allele/">allele model</a></li>
	</ul>
      </div>
      <div class="col-md-8">
	<h1>SimpleCanonicalAllele</h1>
	<h2 id="definition">Definition</h2>

<p>A stable identifier for a SimpleAllele, grouping together the different ways that the allele might be described across different versions of different ReferenceSequences.</p>

<h2 id="scope-and-usage">Scope and Usage</h2>

<p>See also: SimpleAllele, CanonicalAllele</p>

<p>SimpleCanonicalAllele is a CanonicalAllele that is derived from the aggregation of multiple SimpleAllele instances, which each record a particular choice of a variant sequence at a particular contiguous region in a particular ReferenceSequence.</p>

<p>SimpleCanonicalAllele is an abstract class, so no instances of bare SimpleCanonicalAllele can exist; every instance must either be a SimpleNucleotideCanonicalAllele or a SimpleProteinCanonicalAllele.</p>

<p>Like any CanonicalAllele, a SimpleCanonicalAllele has a stable identifier that is independent of updates or additions to the underlying ReferenceSequences.  In addition, a SimpleCanonicalAllele, because it refers to a single contiguous region, may have a LocusIdentifier, that canonically describes that contiguous region, independent of particular references.</p>

<p>While SimpleNucleotideAlleles may be aggregated together into a SimpleNucleotideCanonicalAllele, and SimpleProteinAlleles may be aggregated together into a SimpleProteinCanonicalAllele, SimpleNucleotideAlleles may not be aggregated with SimpleProteinAlleles.  Aggregation of this type would imply identity between these two types of alleles, which is invalid, since multiple nucleotide alleles may lead to the same protein allele, due to codon degeneracy.</p>

<h2 id="exclusions-and-limitations">Exclusions and Limitations</h2>

<p>Given, for example, two nearby SNPs, one technically possible representation would be to write a SimpleAllele that includes the entire reference stretch between them.   While this is not explicitly disallowed by the model, suggested that a more useful approach is to treat this situation with two SimpleAlleles (corresponding to the two SNPs) aggregated into a ComplexNucleotideCanonicalAllele</p>

      </div>
    </div>
  </body>
</html>
